### DISCUSSION TITLE
Multiple servers to the same client? Peer-to-peer planned in future?

### DISCUSSION THREAD

**Discussion by PixelHunter88** [Mimetis/Dotmim.Sync#541] [Q&A] [answered]
Hi Sébastien,

because everything works very well, I would like to reach the next step in my requirements to connect multiple databases with identical schemes.

Hopefully I learned and understand more about the concept behind Dotmim.Sync and I read all relevant issue posts about this question.

I just want to confirm my conclusions:

* For now it is not possible to build a sync structure with multiple servers to the same client, even if the clients never pushes data?

Because ideal for me would be multiple server dbs each having a seperate import db, but some data should be pushed to the same online export db.

* To solve that, it would need one big server which controls all pulls and pushes, because you implemented here a client-server only solution, right?

This would be not easy to handle for me, because it would result easy in a 50+ GB size and I would have to change a lot of processes.

Further I had a 2 day research about the possilities for a peer-to-peer solution, which brought me to MS Sync Framework 2.1, which seems not to be developed any further. So it seems to be not a good choice anymore.

But I wonder why it's not developed any further and why my research for alternatives are surprisingly sparse, even in the commercial sector.

Do you have plans for the future to go in a peer-to-peer direction?

How big you estimate would be the efforts to implement in Dotmim.Sync to a peer-to-peer functionalty?
I would give it both thumbs up and do my best to support that.

Hmm, I see now my questions might be better asked under discussions, sorry for that mistake.

Best Florian

**Comment by Mimetis** [comment, marked as answer]
Hey @PixelHunter88 

## Multiple Servers

It can work, only if the tables from **Server 1** are not the same than tables from **Server 2**.
Working with different scopes can work if you have such topology.

For instance, this **scenario 1** can work:

- Server 1 : 
  - Scope name: S1
  - Tables : TA, TB, TC
- Server 2 : 
  - Scope name: S2
  - Tables: TD, TE, TF

You client will have two scopes (`S1` and `S2`) and 6 tables : `TA`, `TB`, `TC`, `TD`, `TE` and `TF`

And this **scenario 2** will not work

- Server 1 : 
  - Scope name: S1
  - Tables : TA, TB, TC
- Server 2 : 
  - Scope name: S2
  - Tables : TA, TB, TC

> because everything works very well, I would like to reach the next step in my requirements to connect multiple databases with identical schemes

I guess you are in the **scenario 2**, so, unfortunately, it won't work.

_That being said, I never tested this scenario. If you want to take a look on what is feasible or not, let me know_

## Peer to Peer

Indeed, the only solution for having a peer to peer sync, is to use the **MS Sync Framework 2.1**
I've worked a lot with this framework, and, even if it was a great framework, back in the days, it lacks a lot of feature, to be production ready, and since it's not developed anymore, it's not compatible with the last .Net Framework

And the Setup with MS Sync Fx 2.1 is just a nightmare :D

I would have loved to have such feature in `DMS` but it's completely different, and, as far as I know, it won't be possible to integrate such feature in `DMS` unless we start from scratch.

The investment to make a peer to peer sync framework, with the same philosophy as `DMS`, would be too much, too large, comparing to the numbers of people who actually need it, I guess.
